types/src/transaction/mod.rs

```diff
@@ -1002,11 +1011,13 @@ impl TransactionStatus {
             _ => self,
         }
     }
-}
 
-impl From<VMStatus> for TransactionStatus {
-    fn from(vm_status: VMStatus) -> Self {
-        TransactionStatus::from_vm_status(vm_status, true)
+    pub fn from_executed_vm_status(vm_status: VMStatus) -> Self {
+        if vm_status == VMStatus::Executed {
+            TransactionStatus::from_vm_status(vm_status, true, &Features::default())
```

Comment:
The reason we are comfortable passing in the default features is that we are confident all of them are irrelevant for an executed txn? Otherwise this seems sketchy -- even if we disable the feature on-chain, it can still be enabled for this function.

Reply:
gotcha.. Executed is actually "success". In that case, let's refactor so that a bool is passed in, because Features::default() is not that trivial (see the implementation)